### PR TITLE
chore: update set-output usage

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-3708]

### Description

`set-output` and `set-state` are deprecated https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files

### How to test

- check [Draft PR in Picasso](https://github.com/toptal/picasso/pull/3377), there are no deprecation warnings anymore

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/davinci-github-actions/blob/master/_docs/contribution/changeset-guidelines.md) (if needed)

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

[FX-3708]: https://toptal-core.atlassian.net/browse/FX-3708?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: report-missing-changeset/action.yml [hunk 1]
```diff
@@ -27,10 +27,10 @@ runs:
 
     - name: Get changeset status
       id: changeset-info
-      uses: toptal/davinci-github-actions/get-changeset-info@v4.5.0
+      uses: toptal/davinci-github-actions/get-changeset-info@set-output-deprecation
```
Comment (TomasSlama) [inline report-missing-changeset/action.yml:30]: This should be always set to latest version 🤔, but as I recall it is not good practice in GHA, right @dmaklygin ?

Comment (dmaklygin) [reply report-missing-changeset/action.yml:30]: When working with our own GH Actions, it's best to use the latest version available to us. However, it's important to note that any restrictions we encounter are usually related to external GH Actions.

**Review by dmaklygin — APPROVED** [review]
Thank you, good work!

**Review by sashuk — APPROVED** [review]
Looks good, there should be no changes to the way data is passed

**Pull request merged**